ibm/provider/provider.go

```diff
@@ -1120,13 +1121,13 @@ func Provider() *schema.Provider {
 			"ibm_sm_iam_credentials_secret":                                      secretsmanager.AddInstanceFields(secretsmanager.ResourceIbmSmIamCredentialsSecret()),
 			"ibm_sm_username_password_secret":                                    secretsmanager.AddInstanceFields(secretsmanager.ResourceIbmSmUsernamePasswordSecret()),
 			"ibm_sm_kv_secret":                                                   secretsmanager.AddInstanceFields(secretsmanager.ResourceIbmSmKvSecret()),
-			"ibm_sm_configuration_public_certificate_ca_lets_encrypt":            secretsmanager.AddInstanceFields(secretsmanager.ResourceIbmSmConfigurationPublicCertificateCALetsEncrypt()),
-			"ibm_sm_configuration_public_certificate_dns_cis":                    secretsmanager.AddInstanceFields(secretsmanager.ResourceIbmSmConfigurationPublicCertificateDNSCis()),
-			"ibm_sm_configuration_public_certificate_dns_classic_infrastructure": secretsmanager.AddInstanceFields(secretsmanager.ResourceIbmSmConfigurationPublicCertificateDNSClassicInfrastructure()),
-			"ibm_sm_configuration_private_certificate_root_ca":                   secretsmanager.AddInstanceFields(secretsmanager.ResourceIbmSmConfigurationPrivateCertificateRootCA()),
-			"ibm_sm_configuration_private_certificate_intermediate_ca":           secretsmanager.AddInstanceFields(secretsmanager.ResourceIbmSmConfigurationPrivateCertificateIntermediateCA()),
-			"ibm_sm_configuration_private_certificate_template":                  secretsmanager.AddInstanceFields(secretsmanager.ResourceIbmSmConfigurationPrivateCertificateTemplate()),
-			"ibm_sm_configuration_iam_credentials":                               secretsmanager.AddInstanceFields(secretsmanager.ResourceIbmSmConfigurationIamCredentials()),
+			"ibm_sm_public_certificate_configuration_ca_lets_encrypt":            secretsmanager.AddInstanceFields(secretsmanager.ResourceIbmSmPublicCertificateConfigurationCALetsEncrypt()),
```

Comment:
Important note: Once we do a prod release we shouldn' t change the resource/datasource names in next upcominf changes. If we need to change we need Standarad guidelines of Hashicorp https://developer.hashicorp.com/terraform/plugin/sdkv2/best-practices/deprecations